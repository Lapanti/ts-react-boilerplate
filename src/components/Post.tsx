import * as React from 'react';

export interface IPostProps {
    post: IPost;
};

export interface IPost {
    userId: number;
    id: number;
    title: string;
    body: string;
};

const Post: React.StatelessComponent<IPostProps> = ({ post }) => (
    <div className="post">
        <h3 className="post__header">{post.title}</h3>
        <p className="post__body">{post.body}</p>
    </div>
);

export default Post;
