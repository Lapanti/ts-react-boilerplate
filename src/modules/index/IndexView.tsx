import * as React from 'react';
import { Dispatch, Action } from 'redux';
import Post, { IPost } from '../../components/Post';

export interface IIndexProps {
    name: string;
    post?: IPost;
    setName: (n: string) => Action;
    fetchPost: () => Action;
    dispatch: Dispatch<Action>;
}

const IndexView: React.StatelessComponent<IIndexProps> = ({ name, post, setName, fetchPost, dispatch }) => (
    <div className="index">
        <h1 className="index__header">{name}</h1>
        <input className="index__input" type="text" onChange={e => dispatch(setName(e.target.value))} />
        <br />
        <button className="index__button" onClick={() => dispatch(fetchPost())}>
            This button will be styled
        </button>
        {post && <Post post={post} />}
    </div>
);

export default IndexView;
