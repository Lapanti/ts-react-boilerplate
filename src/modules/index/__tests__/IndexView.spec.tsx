import * as React from 'react';
import { shallow } from 'enzyme';
import IndexView from '../IndexView';
import Post, { IPost } from '../../../components/Post';

describe('IndexView', () => {
    const testPost: IPost = { userId: 0, id: 0, title: 'A_TITLE', body: 'A_BODY' };
    const testName = 'A_TEST_NAME';
    const testSetName = jest.fn();
    const testFetchPost = jest.fn();
    const testDispatch = jest.fn();
    const wrapperNoPost = shallow(<IndexView name={testName} setName={testSetName} fetchPost={testFetchPost} dispatch={testDispatch} />);

    it('should render with correct props', () => {
        expect(wrapperNoPost).toMatchSnapshot();
        const wrapperPost = shallow(<IndexView name={testName} post={testPost} setName={testSetName} fetchPost={testFetchPost} dispatch={testDispatch} />);
        expect(wrapperPost).toMatchSnapshot();
    });

    it('should call the correct functions when typing to input field', () => {
        const testValue = 'A_TEST_VALUE';
        wrapperNoPost.find('[type="text"]').simulate('change', { target: { testValue }});
        expect(testDispatch).toBeCalledWith(testSetName(testValue));
    });

    it('should call the correct functions on click', () => {
        wrapperNoPost.find('button').simulate('click');
        expect(testDispatch).toBeCalledWith(testFetchPost());
    });
});
