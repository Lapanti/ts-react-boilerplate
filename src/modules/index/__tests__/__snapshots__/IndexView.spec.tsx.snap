// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexView should render with correct props 1`] = `
<div
  className="index"
>
  <h1
    className="index__header"
  >
    A_TEST_NAME
  </h1>
  <input
    className="index__input"
    onChange={[Function]}
    type="text"
  />
  <br />
  <button
    className="index__button"
    onClick={[Function]}
  >
    This button will be styled
  </button>
</div>
`;

exports[`IndexView should render with correct props 2`] = `
<div
  className="index"
>
  <h1
    className="index__header"
  >
    A_TEST_NAME
  </h1>
  <input
    className="index__input"
    onChange={[Function]}
    type="text"
  />
  <br />
  <button
    className="index__button"
    onClick={[Function]}
  >
    This button will be styled
  </button>
  <Post
    post={
      Object {
        "body": "A_BODY",
        "id": 0,
        "title": "A_TITLE",
        "userId": 0,
      }
    }
  />
</div>
`;
